core/controllers/reader.py

```diff
@@ -329,12 +329,19 @@ def post(self, unused_exploration_id):
         """
         # A domain object representing the old state.
         old_state = exp_domain.State.from_dict(self.payload.get('old_state'))
+        # The id of the exploration to which the state belongs.
+        exp_id = self.payload.get('exp_id')
+        # The version of the exploration when the classifier model was created.
+        exp_version_when_created = self.payload.get('exp_version')
```

Comment:
`exp_version_when_created` is used only when there is need for creating a new classifier model (so that it gets created with the correct `exp_version_when_created`). It is not required if the classifier model exist and is being used only to predict a response.